WordPress/gutenberg
# Add useStateWithHistory hook and use to show a block editor with undo/redo

Related to #53874 

## What?

Undo/Redo has been a common struggle for folks implementing a custom block editor. This PR introduces a `useStateWithHistory` to the compose package. 

Also, I'm adding a new "playground" story to Storybook to show a small block editor using this hook to implement undo/redo.

## Testing Instructions

1- Run storybook `npm run storybook:dev`.
2- Open the "Playground with undo/redo" story.
3- Check that you can use the editor and click undo/redo buttons
